forum/qa-content/qa-ask.js

```diff
@@ -296,3 +296,36 @@ function set_category_description(idprefix)
 		n.innerHTML=desc;
 	}
 }
+
+/*
+ * Feature - Remove unnecessary # from tags, when creating new post (question)
+ */ 
+ ;(function (document)
+ {
+	 'use strict';
+	 
+	 // wait for DOM to load
+	 window.addEventListener('DOMContentLoaded', function()
+	 {
+		// get DOM elements: "tags" input and button "Zadaj pytanie"
+		var tags = document.getElementById('tags');
+		var askQuestionBtn = document.querySelector('input[value="Zadaj pytanie"]');
+		
+		// when user clicks "Zadaj pytanie"
+		askQuestionBtn.addEventListener('click', function(ev)
+		{					
+			// get written tags and split them
+			var allTags =  tags.value.split(' ');
+			
+			// filter tags, accept only those without # and those like "C#"
+			var filteredTags = allTags.filter(function(hash)
+			{
+				if (hash.indexOf('#') < 0 || hash.toLowerCase().indexOf('c#') > -1)

```

Comment:
Hmm.., I assumed that users will not search (and mark as tags) languages with "#" character, because there are any category on forum for languages **other than** C# - except I forgot about category "Inne języki". Ok then, there may be questions and tags that contains languages with "#" other than mentioned C#.
I have looked on Your linked Wiki and what I saw is mostly 10 per 11 languages has "#" as **second character**, excluding "Visual J#" - which, on the other hand, I can't find was searched on Forum (so it can be almost sure, that nobody asks about that language).
Considering that fact and recalling `KISS`principle ( ;) ) I propose to accept only tags, that have not any "#" or have it positioned as a second character in single tag. So for example these tags:
`f# programming# newbie c# h#hash` after filtering will be saved as `f# newbie c# h#hash` within this simple algorithm: `return hash.indexOf('#') < 0 || hash.indexOf('#') === 1`.
I onyl hope nobody put tags like "o#pomocy" :)

I hope You will agree with that solution.